frappe/public/js/frappe/form/workflow.js

```diff
@@ -112,10 +112,12 @@ frappe.ui.form.States = class FormStates {
 								})
 								.then((doc) => {
 									frappe.model.sync(doc);
+									me.frm.script_manager.trigger("after_workflow_action");
```

Comment:
Why is this line being moved? 

It used to execute after refresh, now it will execute before refresh. Doesn't make sense.

Reply:
@rtdany10 Just keep `unfreeze` in finally block? other statements does not make sense in `finally` if there's an exception.

Reply:
Done @surajshetty3416 @ankush